src/libpriv/rpmostree-util.c

```diff
@@ -529,3 +529,74 @@ rpmostree_str_replace (const char  *buf,
 
   return g_regex_replace_literal (regex, buf, -1, 0, new, 0, error);
 }
+
+/* Like realpath(3), but ensures we stay within the given rootfs. We assume the
```

Comment:
Hmm...did you consider my suggestion to do this like:

```
for path in usr/lib/os-release etc/os-release; do
  if fstatat(dfd, path):
    replace_contents_at(dfd, path)
    break
```

It seems to me to be a lot simpler, easier to review and more obviously correct.